static/js/index.js

```diff
@@ -171,8 +174,31 @@ ep_comments.prototype.init = function(){
     saveItemsData: this.saveCommentWithoutSelection.bind(this),
     saveSubItemsData: this.saveRepliesWithoutSelection.bind(this),
   });
+
+  linesChangedListenerDetailed.onLineChanged('*', function(linesChanged) {
+    self._setLineOfChange(linesChanged);
+    self.updateIconsPositionSchedule.padChanged();
+  }, this._getRep());
+
 };
 
+ep_comments.prototype._getRep = function() {
+  var rep;
+  this.ace.callWithAce(function(ace) {
+    rep = ace.ace_getRep();
+  });
+  return rep;
+}
+
+ep_comments.prototype._setLineOfChange = function(linesChanged) {
+  var topLineOfChange = _.min(linesChanged);
+  var lineOfChange = this.textMarkIconsPosition.lineOfChange;
+  var needToUpdateLineOfChange = lineOfChange === undefined || lineOfChange > topLineOfChange;
+  if (needToUpdateLineOfChange) {
+    this.textMarkIconsPosition.lineOfChange = topLineOfChange;
+  }
```

Comment:
Se entendi direito a lógica, aqui vc precisa pegar (1) a linha mais acima do conjunto atual de linhas alteradas; e (2) ver se esse mínimo está acima da linha que eu tinha guardado anteriormente. É isso?

Se for este o caso, vc vai poder simplificar essa função aqui quando for mover o `lineOfChange` do `this.textMarkIconsPosition` para o `this`:

```
ep_comments.prototype._updateMinLineChanged = function(linesChangedOnThisBatch) {
  var topLineChangedOnThisBatch = _.min(linesChangedOnThisBatch);
  this.topLineChanged = _.min(topLineChangedOnThisBatch, this.topLineChanged);
}
```

(Renomeei algumas variáveis e a função para ficar mais claro o que elas fazem)